model/src/lib.rs

```diff
@@ -5,9 +5,24 @@ pub mod version;
 use std::collections::HashMap;
 
 use resource::{Device, Flow, Node, Receiver, ResourceBundle, Sender, Source};
+use tai::TaiTime;
 use tokio::sync::{RwLock, RwLockReadGuard};
 use uuid::Uuid;
 
+macro_rules! modify_resource {
```

Comment:
This is the exact same workflow for all resource types, ideally this would be done without a macro but i didn't want to change the interface too much, so i stuck to the style of the author